packages/db/src/Page/MaterialPage.tsx

```diff
@@ -0,0 +1,224 @@
+import {Region, Servant, ClassName, Item, Entity} from "@atlasacademy/api-connector";
+import {AxiosError} from "axios";
+import React from "react";
+import {Tab, Tabs} from "react-bootstrap";// Col, Row,
+import {withRouter} from "react-router";
+import {RouteComponentProps} from "react-router-dom";
+import Api from "../Api";
+import MaterialUsageBreakdown from "../Breakdown/MaterialUsageBreakdown";
+import ItemIcon from "../Component/ItemIcon";
+import ClassIcon from "../Component/ClassIcon";
+import DataTable from "../Component/DataTable";
+import ErrorStatus from "../Component/ErrorStatus";
+import RawDataViewer from "../Component/RawDataViewer";
+import Loading from "../Component/Loading";
+import Manager from "../Setting/Manager";
+import {MaterialUsageData} from "./Material/MaterialUsageData";
+
+interface IProps extends RouteComponentProps {
+    region: Region;
+    id: number;
+    tab?: string;
+}
+
+interface IState {
+    error?: AxiosError;
+    loading: boolean;
+    id: number;
+    servants: Servant.Servant[];
+    material?: Item.Item;
+}
+
+class MaterialPage extends React.Component<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            loading: true,
+            id: this.props.id,
+            servants: [],
+        };
+    }
+
+    componentDidMount() {
+        Manager.setRegion(this.props.region);
+        this.loadItem();
+    }
+
+    itemIsMaterial(material: Item.Item) {
+        if (material.type === "skillLvUp" || material.type === "tdLvUp") {
+            if (material.id > 6000 && material.id < 6208) return true; // Matches Gems
+            if (material.id > 6500 && material.id < 6600) return true; // Matches Mats
+            if (material.id > 7000 && material.id < 7108) return true; // Matches Statues
+        }
+        return false;
+    }
+
+    async loadItem() {
+        try {
+            let [servants, material] = await Promise.all<Servant.Servant[], Item.Item>([
+                Api.servantListNice(),
+                Api.item(this.state.id)
+            ]);
+
+            if (!this.itemIsMaterial(material)) {
+                throw Error(`Item ${this.state.id} is not a non-event Skill or Ascension Material.`);
+            }
+
+            this.setState({
+                loading: false,
+                servants,
+                material
+            });
+        } catch (e) {
+            this.setState({
+                error: e
+            });
+        }
+    }
+
+    private reduceMaterials(materials: Object) {
```

Comment:
@minhducsun2002 I probably would look into this further

Reply:
Done.